app/controllers/api/v2/hosts_controller.rb

```diff
@@ -280,8 +280,13 @@ def boot
       param :facts, Hash,      :required => true, :desc => N_("hash containing the facts for the host")
       param :certname, String, :desc => N_("optional: certname of the host")
       param :type, String,     :desc => N_("optional: the STI type of host to create")
-
       def facts
+        unless Setting[:enable_facts_storage]
+          render_error 'custom_error', :status => :unprocessable_entity,
+                       :locals => { :message => "Fact Values feature is disabled, you can enable it in Settings > General" }
+          return
+        end
```

Comment:
The idea is not to throw any error when facts/reports storage is disabled, but to return 200 HTTP OK but doing nothing.

Reply:
But @ekohl wanted to return error status code. So how it should be?